### PR TITLE
Allow transforming the DB2 JDBC driver to Jakarta APIs during Augmentation

### LINKED ISSUE #27066: Jakarta - EE 9 - Native - Db2 driver depends on javax.transaction
The Db2 JDBC driver depends on some `javax.transaction` classes that have been moved to the `jakarta` root package.
I don't know if they are shipping a Jakarta-compatible version somewhere.

```
Caused by: org.graalvm.compiler.java.BytecodeParser$BytecodeParserError: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: javax.transaction.TransactionManager. This error is reported at image build time because class com.ibm.db2.jcc.t2zos.T2zosConnection is registered for linking at image build time by command line
	at parsing com.ibm.db2.jcc.t2zos.T2zosConnection.k(T2zosConnection.java:4970)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.throwParserError(BytecodeParser.java:2506)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.throwParserError(SharedGraphBuilderPhase.java:105)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3367)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.handleBytecodeBlock(BytecodeParser.java:3319)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3164)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.build(BytecodeParser.java:1138)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1030)
	at jdk.internal.vm.compiler/org.graalvm.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:84)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:79)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.run(Phase.java:49)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:261)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:42)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:38)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecode(AnalysisParsedGraph.java:135)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:685)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:168)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:343)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:93)
```

### PR DESCRIPTION
**Pull request by Sanne**
I've spent quite some time trying to fix this in other ways, including trying to add some magic to GraalVM, but ultimately it seems that using Eclipse Transformer during Augmentation was much simpler than I expected.

I suppose this could evolve into a more reusable build item, but I'll avoid making this too generic for now, and hope we won't need this too often.

Fixes #27066

### CODE REVIEW

File: extensions/jdbc/jdbc-db2/deployment/pom.xml [hunk 1]
```diff
@@ -48,6 +48,11 @@
             <artifactId>assertj-core</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.eclipse.transformer</groupId>
+            <artifactId>org.eclipse.transformer</artifactId>
+            <version>0.5.0</version>
+        </dependency>
```
Comment (gastaldi) [inline extensions/jdbc/jdbc-db2/deployment/pom.xml:55]: Move this to our BOM?

File: extensions/jdbc/jdbc-db2/deployment/src/main/java/io/quarkus/jdbc/db2/deployment/JakartaEnablement.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.BytecodeTransformerBuildItem;
+
+/**
+ * The DB2 driver is compiled using references to classes in the javax.transaction
+ * package; we need to transform these to fix compatibility with jakarta.transaction.
+ * We do this by leveraging the Eclipse Transformer project during Augmentation, so
+ * that end users don't need to bother.
+ *
+ * @author Sanne Grinovero
```
Comment (gsmet) [inline extensions/jdbc/jdbc-db2/deployment/src/main/java/io/quarkus/jdbc/db2/deployment/JakartaEnablement.java:32]: You know we don't do the `@author` tag here :).

**Review by gsmet — COMMENTED** [review]
Very interesting idea. It's probably the best we can do for DB2 until they release a driver. I added a small comment.

I'm surprised it's so simple :).

File: extensions/jdbc/jdbc-db2/deployment/pom.xml (hunk 1 above)
Comment (gsmet) [reply extensions/jdbc/jdbc-db2/deployment/pom.xml:55]: I think I would rather not enforce it for now. We can revisit if we have more modules using it but I would rather let projects use the version they want given we only use a small capability of it.

Comment (Sanne) [reply extensions/jdbc/jdbc-db2/deployment/pom.xml:55]: Right, I was in doubt about this and leaning for the less disruptive approach keeping it here. Happy to revisit when there's any need.

File: extensions/jdbc/jdbc-db2/deployment/src/main/java/io/quarkus/jdbc/db2/deployment/JakartaEnablement.java (hunk 2 above)
Comment (Sanne) [reply extensions/jdbc/jdbc-db2/deployment/src/main/java/io/quarkus/jdbc/db2/deployment/JakartaEnablement.java]: sorry :) fixed

**Comment by Sanne** [comment]
> Very interesting idea. It's probably the best we can do for DB2 until they release a driver. I added a small comment.
> 
> I'm surprised it's so simple :).

I was too :) Quite pleased with the outcome. You should have seen all other attempts - dozens of obcure and unmainaintable classes :D

**Review by gsmet — APPROVED** [review]

**Pull request merged**